homework/index.js

```diff
@@ -16,31 +30,116 @@
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function addRow(tableBody, description, className, elementContent) {
+    const tableRow = createAndAppend('tr', tableBody, { class: 'flex-div' });
+    createAndAppend('td', tableRow, {
+      text: description,
+      class: className,
+    });
+    createAndAppend('td', tableRow, {
+      text: elementContent,
+    });
+    return tableRow;
+  }
+
+  function renderRepository(tableDiv, repository) {
+    tableDiv.innerHTML = '';
+
+    const table = createAndAppend('table', tableDiv);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository:', 'td-header', ' ');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    addRow(tbody, 'Description:', 'description-td td-header', repository.description);
+    addRow(tbody, 'Forks:', 'td-header', repository.forks_count);
+    addRow(tbody, 'Updated:', 'td-header', new Date(repository.updated_at).toLocaleString());
+  }
+
+  function renderContributions(url, contributesDiv) {
+    fetchJSON(url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+
+      contributesDiv.innerHTML = '';
+      createAndAppend('h1', contributesDiv, { class: 'contributes-header', text: 'Contributions' });
+
+      const ul = createAndAppend('ul', contributesDiv);
+
+      Object.keys(contributors).forEach(contributor => {
+        const hyperlink = createAndAppend('a', ul, {
+          href: contributors[contributor].html_url,
+          target: '_blank',
+        });
+
+        const listItem = createAndAppend('li', hyperlink, { class: 'list flex-div' });
+        createAndAppend('img', listItem, {
+          src: contributors[contributor].avatar_url,
+          alt: `${contributors[contributor].login} photo`,
+        });
+
+        const contributorInfoDiv = createAndAppend('div', listItem, {
+          class: 'contributor-info flex-div',
+        });
+        createAndAppend('p', contributorInfoDiv, {
+          class: 'contributorsName',
+          text: contributors[contributor].login,
+        });
+        createAndAppend('p', contributorInfoDiv, {
+          text: contributors[contributor].contributions,
+          class: 'badge',
+        });
+      });
+    });
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, { text: repository.name, value: index });
     });
-    return elem;
   }
 
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
+    const root = document.getElementById('root');
+
+    fetchJSON(url, (err, data) => {
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
```

Comment:
Why did you take out the return? By taking it out you now stumble into code that you should not be executing in case of errors. It will produce this error in the console:

```
index.js:123 Uncaught TypeError: Cannot read property 'sort' of undefined
    at index.js:123
    at XMLHttpRequest.xhr.onload (index.js:26)
```